src/internal/components/pluggableVisualizations/pivotTable/sortItemsHelpers.ts

```diff
@@ -0,0 +1,191 @@
+    buckets: VisualizationObject.IBucket[],
+) => flatMap(buckets.filter(b => b.localIdentifier === bucketId), i => i.items) as T[];
+export const getMeasures = bucketItemGetter<VisualizationObject.IMeasure>(BucketNames.MEASURES);
+export const getRows = bucketItemGetter<VisualizationObject.IVisualizationAttribute>(BucketNames.ATTRIBUTE);
+export const getColumns = bucketItemGetter<VisualizationObject.IVisualizationAttribute>(BucketNames.COLUMNS);
+
+export function adaptMdObjectSortItemsToPivotTable(
+    originalSortItems: AFM.SortItem[],
+    buckets: VisualizationObject.IBucket[],
+): AFM.SortItem[] {
+    const measureLocalIdentifiers = getMeasures(buckets).map(measure => measure.measure.localIdentifier);
+    const rowAttributeLocalIdentifiers = getRows(buckets).map(
+        rowAttribute => rowAttribute.visualizationAttribute.localIdentifier,
+    );
+    const columnAttributeLocalIdentifiers = getColumns(buckets).map(
+        columnAttribute => columnAttribute.visualizationAttribute.localIdentifier,
+    );
+
+    return adaptSortItemsToPivotTable(
+        originalSortItems,
+        measureLocalIdentifiers,
+        rowAttributeLocalIdentifiers,
+        columnAttributeLocalIdentifiers,
+    );
+}
+
+const isAttributeSortItemVisible = (_sortItem: AFM.IAttributeSortItem, _filters: IBucketFilter[]): boolean =>
+    true;
+
+const isMeasureSortItemMatchedByFilter = (
+    sortItem: AFM.IMeasureSortItem,
+    filter: IAttributeFilter,
+): boolean =>
+    filter.selectedElements.some(selectedElement =>
+        sortItem.measureSortItem.locators.some(
+            locator =>
+                !AFM.isMeasureLocatorItem(locator) &&
+                locator.attributeLocatorItem.element === selectedElement.uri,
+        ),
+    );
+
+const isMeasureSortItemVisible = (sortItem: AFM.IMeasureSortItem, filters: IBucketFilter[]): boolean =>
+    filters.reduce((isVisible, filter) => {
+        if (isAttributeFilter(filter)) {
+            const shouldBeMatched = !filter.isInverted;
+            return isVisible && shouldBeMatched === isMeasureSortItemMatchedByFilter(sortItem, filter);
+        }
+        return isVisible;
+    }, true);
+
+export const isSortItemVisible = (sortItem: AFM.SortItem, filters: IBucketFilter[]): boolean =>
+    AFM.isAttributeSortItem(sortItem)
+        ? isAttributeSortItemVisible(sortItem, filters)
+        : isMeasureSortItemVisible(sortItem, filters);
+
+export function addDefaultSort(
+    sortItems: AFM.SortItem[],
+    filters: IBucketFilter[],
+    rowAttributes: IBucketItem[],
+    previousRowAttributes?: IBucketItem[],
+): AFM.SortItem[] {
+    // cannot construct default sort without a row
+    if (rowAttributes.length < 1) {
+        return sortItems;
+    }
+
+    // detect custom sort
+    const firstRow = rowAttributes[0];
+    const previousFirstRow = previousRowAttributes && previousRowAttributes[0];
+    const hasVisibleCustomSort = sortItems.some(sortItem => {
```

Comment:
😅

Reply:
It's strange but since we just moved this code to the new place I would like to keep it the same.